SegM8/display-text/patch.cpp

```diff
@@ -0,0 +1,23 @@
+
+struct State {
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+    auto ext = getValue<input_DEV>(ctx);
+    auto text = getValue<input_TEXT>(ctx);
+    auto position = getValue<input_POS>(ctx);
+    auto width = getValue<input_WDTH>(ctx);
+
+    if (isSettingUp())
+        emitValue<output_DEVU0027>(ctx, ext);
+
+    if(isInputDirty<input_DO>(ctx)) {
+        char* buf = new char(length(text)+1);
```

Comment:
`char* buf = new char(length(text)+1);` 

This would not work. The length of the string and its memory usage are different. For example try to display a const string node `11.11` on boot. It will hang the MCU. But, it will work with tweak string.

change to:
```
char buf[length(text) + 1];
dump(text, buf);
ext->chain->display(buf, position, width, SEGM8_ALIGN_LEFT);
```